mindsdb/utilities/security.py

```diff
@@ -31,3 +31,28 @@ def clear_filename(filename: str) -> str:
     for c in badchars:
         filename = filename.replace(c, '')
     return filename
+
+
+# TODO: make this list configurable in config.json
+ALLOWED_DOMAINS = [
+    "s3.amazonaws.com",
+    "drive.google.com",
+    "blob.core.windows.net",
+    "dropbox.com",
+    "githubusercontent.com",
+    "onedrive.live.com"
+]
+
+
+def is_allowed_url(url):
+    """
+    Checks if the provided URL is from an allowed host.
+
+    This function parses the URL and checks the network location part (netloc)
+    against a list of allowed hosts.
+
+    :param url: The URL to check.
+    :return bool:  True if the URL is from an allowed host, False otherwise.
+    """
+    parsed_url = urlparse(url)
```

Comment:
May be worth to add `.lower()` here.
Also `www` may be at the start if url, but that is rare case